wire-ios/Wire-iOS/Sources/Authentication/Helpers/RegistrationAnalyticsTracker.swift

```diff
@@ -16,33 +16,75 @@
 // along with this program. If not, see http://www.gnu.org/licenses/.
 //
 
+import UIKit
+import WireAnalytics
 import WireAuthenticationAPI
 import WireFoundation
+import WireLogging
+import WireSyncEngine
 
-struct RegistrationAnalyticsTracker: RegistrationAnalyticsTrackerProtocol {
+final class RegistrationAnalyticsTracker: RegistrationAnalyticsTrackerProtocol {
 
+    private var analyticsService: AnalyticsService?
     private var analyticsTracker: (any AnalyticsEventTrackerProtocol)?
+    private let userDefaults: UserDefaults
+    private let logger: WireLogger
 
-    init() {}
+    init(
+        analyticsServiceConfiguration: AnalyticsServiceConfiguration?,
+        countlyProvider: @escaping () -> any CountlyProtocol,
+        userDefaults: UserDefaults
+    ) {
+        self.analyticsService = analyticsServiceConfiguration.map { config in
+            AnalyticsService(
+                config: CountlyConfiguration(appKey: config.secretKey, host: config.serverHost),
+                deviceModel: UIDevice.current.model,
+                osVersion: UIDevice.current.systemVersion,
+                countlyProvider: countlyProvider
+            )
+        }
+        self.userDefaults = userDefaults
+        self.logger = .authentication
+    }
+
+    var currentDeviceID: String? {
+        analyticsService?.currentDeviceID
+    }
 
+    @MainActor
     func setUp() {
-        fatalError("WPB-17530")
+        do {
+            let analyticsUser = createAnalyticsUserIfNeeded()
+            try enableAnalytics(user: analyticsUser)
+        } catch {
+            logger.error("Can't set up analytics during personal account registration")
+        }
     }
 
     func tearDown() {
-        fatalError("WPB-17530")
+        do {
+            try disableAnalytics()
+        } catch {
+            logger.error("Can't disable analytics during personal account registration")
+        }
     }
 
     func trackPersonalAccountCreationStart() {
-        analyticsTracker?.trackEvent(.Registration.accountSetupStep0)
+        if let analyticsTracker {
+            analyticsTracker.trackEvent(.Registration.accountSetupStep0)
+        }
     }
 
     func trackPersonalAccountCreationReachedTermsOfUseConfirmation() {
-        analyticsTracker?.trackEvent(.Registration.accountSetupStep1)
+        if let analyticsTracker {
+            analyticsTracker.trackEvent(.Registration.accountSetupStep1)
+        }
     }
 
     func trackPersonalAccountCreationReachedVerificationCode() {
-        analyticsTracker?.trackEvent(.Registration.accountSetupStep2)
+        if let analyticsTracker {
+            analyticsTracker.trackEvent(.Registration.accountSetupStep2)
+        }
```

Comment:
nit-pic: don't unwrap the value, or be consisten with trackPersonalAccountCreationFailedCodeVerification